### PR TITLE
Add missing docs, package info and bugfixes

### PR DESCRIPTION
**Pull request by simonporter**
Add package info for creating nuget packages including documentation.
Add missing XML docs on public classes.
Fix bug with RaiseEvent calling create orchestration causing unneeded jumpstart entry.
Always set mindatetime for storage instance store, not just for emulator.

### CODE REVIEW

File: src/DurableTask.Core/OrchestrationStateInstanceFilter.cs [hunk 1]
```diff
@@ -13,16 +13,33 @@
...
+        /// </summary>
         public string InstanceId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the ExecutionId for the filter
+        /// </summary>
         public string ExecutionId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the match type of either statrts with or exact match for the filter
```
Comment (rajeevmv) [inline src/DurableTask.Core/OrchestrationStateInstanceFilter.cs]: >statrts [](start = 50, length = 7)

typo

File: src/DurableTask.Core/OrchestrationStateNameVersionFilter.cs [hunk 2]
```diff
@@ -13,9 +13,19 @@
 
 namespace DurableTask.Core
 {
+    /// <summary>
+    /// Filter for Orchestrationname and version
```
Comment (rajeevmv) [inline src/DurableTask.Core/OrchestrationStateNameVersionFilter.cs]: split

File: src/DurableTask.Core/OrchestrationStateTimeRangeFilter.cs [hunk 3]
```diff
@@ -15,10 +15,24 @@ namespace DurableTask.Core
 {
     using System;
 
+    /// <summary>
+    /// Filter for Orchestration time range on a timerange type
```
Comment (rajeevmv) [inline src/DurableTask.Core/OrchestrationStateTimeRangeFilter.cs]: + State?

File: src/DurableTask.Core/Common/Utils.cs [hunk 4]
```diff
@@ -25,15 +25,24 @@ namespace DurableTask.Core.Common
...
 
+    /// <summary>
+    /// Utility Methods
+    /// </summary>
     public static class Utils
     {
         const int FullGzipHeaderLength = 10;
 
+        /// <summary>
+        /// Gets a safe maximim datetime value that accounts for timezone
```
Comment (pugup) [inline src/DurableTask.Core/Common/Utils.cs]: maximum

Comment (rajeevmv) [inline src/DurableTask.Core/Common/Utils.cs]: >maximim [](start = 24, length = 7)

typo

File: src/DurableTask.Emulator/LocalOrchestrationService.cs [hunk 5]
```diff
@@ -50,6 +50,10 @@ public class LocalOrchestrationService : IOrchestrationService, IOrchestrationSe
 
         ConcurrentDictionary<string, TaskCompletionSource<OrchestrationState>> orchestrationWaiters;
 
+        /// <summary>
+        ///     Creates a new instance of the LocalOrchestrationService with default settings
+        /// </summary>
```
Comment (rajeevmv) [inline src/DurableTask.Emulator/LocalOrchestrationService.cs]: Extra line

File: src/DurableTask.ServiceBus/Tracking/AzureTableInstanceStore.cs [hunk 6]
```diff
@@ -47,11 +47,8 @@ public AzureTableInstanceStore(string hubName, string tableConnectionString)
         {
             this.tableClient = new AzureTableClient(hubName, tableConnectionString);
 
-            // Workaround a bug in the Storage Emulator that throws exceptions for any date < 1600 so DateTime.Min cannot be used
-            if (tableConnectionString.Contains("UseDevelopmentStorage=true"))
-            {
-                DateTimeUtils.SetMinDateTimeForStorageEmulator();
-            }
+            // Workaround an issue with Storage that throws exceptions for any date < 1600 so DateTime.Min cannot be used
```
Comment (rajeevmv) [inline src/DurableTask.ServiceBus/Tracking/AzureTableInstanceStore.cs]: > // Workaround an issue with Storage that throws exceptions for any date < 1600 so DateTime.Min cannot be used [](start = 11, length = 110)

Is this comment still relevant?

**Comment by rajeevmv** [comment]
:shipit:

File: src/DurableTask.ServiceBus/Tracking/AzureTableInstanceStore.cs (hunk 6 above)
Comment (simonporter) [inline src/DurableTask.ServiceBus/Tracking/AzureTableInstanceStore.cs]: Yes this is specific to Azure Tables, this code was only enabled for the emulator but the newer storage SDK also needs this

File: src/DurableTask.Core/Common/Utils.cs [hunk 7]
```diff
@@ -57,6 +69,9 @@ public static void WriteObjectToStream(Stream objectStream, object obj)
             objectStream.Position = 0;
         }
 
+        /// <summary>
+        /// Writes the supplied string input to a MemoryStream, optionaly compressing the string
```
Comment (pugup) [inline src/DurableTask.Core/Common/Utils.cs]: nit: optionally

should probably mention that it returns the said MemoryStream

File: src/DurableTask.Core/FrameworkConstants.cs [hunk 8]
```diff
@@ -15,50 +15,147 @@ namespace DurableTask.Core
...
+        /// The default number of orchestration dispatchers
+        /// </summary>
         public const int OrchestrationDefaultDispatcherCount = 1;
+
+        /// <summary>
+        /// The default max concurrent orchestration work items
+        /// </summary>
         public const int OrchestrationDefaultMaxConcurrentItems = 100;
 
         // task activity dispatcher default constants
```
Comment (pugup) [inline src/DurableTask.Core/FrameworkConstants.cs]: nit: perhaps add a newline

File: src/DurableTask.Core/OrchestrationStateInstanceFilter.cs [hunk 9]
```diff
@@ -13,16 +13,33 @@
 
 namespace DurableTask.Core
 {
+    /// <summary>
+    /// Filter for Orchestration instance filter
+    /// </summary>
     public class OrchestrationStateInstanceFilter : OrchestrationStateQueryFilter
     {
+        /// <summary>
+        ///     Creates a new instance of the OrchestrationStateInstanceFilter with default settings
```
Comment (pugup) [inline src/DurableTask.Core/OrchestrationStateInstanceFilter.cs]: nit: comment indentation here and the rest of the code doesn't match. similarly, in a few other places below

**Pull request merged**